src/components/checkbox.js

```diff
@@ -18,17 +18,21 @@ const styles = makeStyles({
   },
 })
 
-const Checkbox = ({ classes, label, checked, defaultChecked, indeterminate, onChange, inputProps, ...rest }) => {
+const Checkbox = ({ classes, label, checked, defaultChecked, indeterminate, onChange, inputProps, showLabel, ...rest }) => {
   const [check, setCheck] = useState({ label, checked })
 
+  useEffect(() => {
+    setCheck({ label, checked })
+  }, [label, checked])
```

Comment:
why is this useEffect needed?

Reply:
it is only initializing it with the 'check' prop but not updating it when 'check' pro has changed so I needed that to make sure it gets update because I am manipulating check prop by click outside of checkbox component